src/metabase/upload.clj

```diff
@@ -763,5 +759,6 @@
     - uploads are enabled
   Otherwise based_on_upload is nil."
   [cards]
-  (let [models-by-id (m/index-by :id (model-hydrate-based-on-upload (filter :dataset cards)))]
```

Comment:
This naming threw me, given the long discussion we had against using this naming schema 😆

Reply:
Ha, maybe this is a different scenario where `x-by-y` adds to readability. `models-by-id` at least implies that it is a map of a _model_ ID to a model (maybe it should rather be `model-by-id`), as opposed to card ID to a model. Because the name reminds me of how it was created: either `m/index-by` or `clojure.core/group-by`. But I think `id->model` is also fine.

The use of `card->model` here is a little surprising for me though. The english description of this function based on its behaviour would be "Returns a hydrated model if a card is a model". Yet to me the name `card->model` implies that there is one model per card. Not an opinion I hold strongly though

Reply:
How about `card->maybe-model` then?

Reply:
What's the point?

Reply:
Ah, this is about mixing abstraction levels inside an `or`. I repeat the point here: introducing extra names comes at a cost because they can be misleading and will make your eyes jump around if you have to understand them.

Reply:
`card->maybe-model` isn't misleading, so I'm happy with that name

Reply:
Also happy if you want to inline it, I felt it helped a little bit with the density of the code but not attached.

Reply:
I like the rule. I don't mind that much with `card->maybe-model`. Apologies for splitting hairs again on naming, maybe why I like inlining things is because I'm so critical of names

Reply:
Don't miss naming day in the gym! In the mean time I'll be doing some "reading code that is more dense than I'm used to" reps.

Reply:
oh you should be aware of another reason I don't like names: I have an aversion to newlines because I work without an external monitor and hate scrolling :D

Reply:
Is it at least 16"? 😆

I feel so blind when I'm just on one small screen for pairing, for sure following references is less painful when they open in a split pane.

Reply:
Yes, 16. Before metabase it was even smaller.